File: scripts/dev/skill-runtime-probe/soul/skills/skill-runtime-probe/scripts/probe.ts
// Sandbox contract: arguments arrive as an Artifact, the result must be written to
// $TULIP_OUTPUT_DIR/result.json, and the container has no network. The runtime only strips
// TypeScript types, so this file must stay erasable syntax.
import { readFileSync, writeFileSync } from "node:fs";
import { join } from "node:path";

interface ProbeArguments {
  readonly message?: string;
}

const inputDir = process.env.TULIP_INPUT_DIR ?? "";
const outputDir = process.env.TULIP_OUTPUT_DIR ?? "";

const args = JSON.parse(readFileSync(join(inputDir, "0-input.json"), "utf8")) as ProbeArguments;
const message = args.message ?? "no message";

process.stderr.write(`probe.ts received: ${message}\n`);

writeFileSync(
  join(outputDir, "result.json"),
  JSON.stringify({
    ok: true,
    runtime: "typescript",
    interpreter: `Node ${process.version}`,
    echoed: message,
  })
);

File: scripts/dev/skill-runtime-probe/soul/skills/skill-runtime-probe/scripts/probe.py
#!/usr/bin/env python3
"""Sandbox contract: arguments arrive as an Artifact, the result must be written to
$TULIP_OUTPUT_DIR/result.json, and the container has no network."""

import json
import os
import platform
import sys

with open(os.path.join(os.environ["TULIP_INPUT_DIR"], "0-input.json"), encoding="utf-8") as handle:
    arguments = json.load(handle)

message = arguments.get("message") or "no message"
print(f"probe.py received: {message}", file=sys.stderr)

with open(
    os.path.join(os.environ["TULIP_OUTPUT_DIR"], "result.json"), "w", encoding="utf-8"
) as handle:
    json.dump(
        {
            "ok": True,
            "runtime": "python",
            "interpreter": f"Python {platform.python_version()}",
            "echoed": message,
        },
        handle,
    )
